src/javascript/class-scanner.ts

```diff
@@ -838,10 +838,10 @@ export function extractPropertiesFromClass(
 
     for (const val of accessors.values()) {
```

Comment:
Hm, there's some other logic here that it would be nice to also have in the behaviors code, like getting `readOnly` correct. What about a method that encapsulates everything with an API like:

```javascript
for (const prop of propertiesFromGettersAndSetters(astNode.body.body)) {
  properties.set(prop.name, prop);
}
```

Reply:
hilariously @rictic it looks like this piece of code was fubar all along too. the accessors loop is _inside_ the body loop... we've been doing an iteration of the accessors per method all along /facepalm